spring-data-cassandra/src/main/java/org/springframework/data/cassandra/config/CqlSessionFactoryBean.java

```diff
@@ -450,11 +451,20 @@ public void afterPropertiesSet() {
 
 		this.session = buildSession(sessionBuilder);
 
-		executeCql(getStartupScripts().stream(), this.session);
-		performSchemaAction();
+		try {
+			SchemaRefreshUtils.withDisabledSchema(this.session, () -> {
+				executeCql(getStartupScripts().stream(), this.session);
+				performSchemaAction();
+			});
+		} catch (RuntimeException e) {
+			throw e;
+		} catch (Exception e) {
+			throw new IllegalStateException("Unexpected checked exception thrown", e);
+		}
 
-		this.systemSession.refreshSchema();
-		this.session.refreshSchema();
+		if (this.systemSession.isSchemaMetadataEnabled()) {
```

Comment:
Why is the refresh for the actual session removed?

Reply:
It actually happens inside `SchemaRefreshUtils.withDisabledSchema`. It isn't obvious from the PR itself, but from the javadoc of `session.setSchemaMetadataEnabled`:
> If calling this method re-enables the metadata (that is, #isSchemaMetadataEnabled() was false before, and becomes true as a result of the call), a refresh is also triggered.

In other words, if schema refresh becomes re-enabled as a result then a refresh will be triggered.

There is one difference though - previously we'd call `refreshSchema` which is sync and blocking while the schema refresh here is more equivalent to calling `Session#refreshSchemaAsync` and discarding the returned `CompletionStage`. I could update the new code here to wait on the returned future but I don't think it's necessary, let me know if you think otherwise!